common/ringpop/ringpop.go

```diff
@@ -116,7 +117,15 @@ func (factory *ringpopFactory) GetMembershipMonitor() (membership.Monitor, error
 func (factory *ringpopFactory) getMembership() (membership.Monitor, error) {
 	var err error
 	factory.monOnce.Do(func() {
-		if rp, err := ringpop.New("temporal", ringpop.Channel(factory.getTChannel()), ringpop.AddressResolverFunc(factory.broadcastAddressResolver)); err != nil {
+		currentClusterMetadata, err := factory.metadataManager.GetCurrentClusterMetadata(context.Background())
```

Comment:
do we need some retry here? also the context needs timeout?

Reply:
Also need to mark the context as background persistence call: https://github.com/temporalio/temporal/blob/master/common/headers/caller_info.go#L41

Reply:
The client is wrapped with retry. I added a timeout of 10s and include the system background header.